cookiecutter/config.py

```diff
@@ -14,7 +14,34 @@
 
 logger = logging.getLogger(__name__)
 
-USER_CONFIG_PATH = os.path.expanduser('~/.cookiecutterrc')
+
+def _expand_path(path):
+    """Expand both environment variables and user home in the given path."""
+    path = os.path.expandvars(path)
+    path = os.path.expanduser(path)
+    return path
+
+
+def _find_first_path(paths):
+    """Find the first existing path from a list."""
+    logger.debug("_find_first_path: %s", paths)
+    for path in paths:
+        if path:
+            path = _expand_path(path)
+            logger.debug("Trying to load: %s", path)
+            if os.path.exists(path):
+                return path
+    return None
+
+
+USER_CONFIG_PATH = _find_first_path(
```

Comment:
Personally, I do not like execution of slightly more complex code on initialization time. This  function should be called in `merge_configs` below.